activemodel/lib/active_model/validations/numericality.rb

```diff
@@ -6,7 +6,7 @@ class NumericalityValidator < EachValidator # :nodoc:
       CHECKS = { greater_than: :>, greater_than_or_equal_to: :>=,
                  equal_to: :==, less_than: :<, less_than_or_equal_to: :<=,
                  odd: :odd?, even: :even?, other_than: :!= }.freeze
-
+      CHECKS_IGNORING_NIL = %i(greater_than greater_than_or_equal_to equal_to less_than less_than_or_equal_to)
```

Comment:
I think we can remove `equal_to` from the ignore list

Reply:
Well, I think you're right and I should remove it - since it does not compare numbers it won't crash in a way described above. Also, it would cause a following confusing behavior:

```ruby
class Smth
  include ActiveModel::Validations
  validates :field_one, numericality: { allow_nil: true }
  validates :field_two, numericality: { allow_nil: true, equal_to: :field_one }
end

Smth.new(field_two: 1).valid? # => true, but why? these two are not equal
```

Removing that away, thanks for the review @bogdanvlviv